Realm/RLMQueryUtil.mm

```diff
@@ -877,6 +878,23 @@ void process_or_group(Query &query, id array, Func&& func) {
     return RLMStringDataWithNSString(value);
 }
 
+template <>
+Decimal128 convert<Decimal128>(id value) {
+    return RLMObjcToDecimal128(value);
+}
+
+template <>
+ObjectId convert<ObjectId>(id value) {
+    // FIXME: extract this somewhere and share with object creation
```

Comment:
Right, shouldn't this also account for number types?

Reply:
What conversion would make sense for number to objectid?

Reply:
Going to resolve this– can't remember the context of why I put this here.